Periodic_2_triangulation_2/include/CGAL/Periodic_2_Delaunay_triangulation_2.h

```diff
@@ -1120,15 +1121,27 @@ bool
 Periodic_2_Delaunay_triangulation_2<Gt, Tds>::
 remove_single_vertex(Vertex_handle v, const Offset &v_o)
 {
-  static int maxd = 30;
-  static std::vector<Face_handle> f(maxd);
-  static std::vector<int> i(maxd);
-  static std::vector<Vertex_handle> w(maxd);
-  static std::vector<Offset> offset_w(maxd);
+  struct Static_data{
```

Comment:
That class, used as a template parameter, cannot be a nested class, in C++03.

https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.10-Ic-28/Periodic_2_triangulation_2/TestReport_gimeno_Debian-Stable.gz
```
In file included from /mnt/testsuite/include/CGAL/Modular_arithmetic/Residue_type.h:25:0,
                 from /mnt/testsuite/include/CGAL/Residue.h:31,
                 from /mnt/testsuite/include/CGAL/Modular_traits.h:24,
                 from /mnt/testsuite/include/CGAL/int.h:30,
                 from /mnt/testsuite/include/CGAL/number_type_basic.h:64,
                 from /mnt/testsuite/include/CGAL/basic.h:44,
                 from /mnt/testsuite/include/CGAL/Periodic_2_offset_2.h:23,
                 from /mnt/testsuite/include/CGAL/Periodic_2_triangulation_traits_2.h:35,
                 from /home/cgal_tester/build/src/cmake/platforms/Debian-Stable/test/Periodic_2_triangulation_2/include/./types.h:10,
                 from /home/cgal_tester/build/src/cmake/platforms/Debian-Stable/test/Periodic_2_triangulation_2/test_p2t2_delaunay_insert_degenerate.cpp:5:
/mnt/testsuite/include/CGAL/Periodic_2_Delaunay_triangulation_2.h: In instantiation of 'bool CGAL::Periodic_2_Delaunay_triangulation_2<Gt, Tds>::remove_single_vertex(CGAL::Periodic_2_Delaunay_triangulation_2<Gt, Tds>::Vertex_handle, const Offset&) [with Gt = CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>; Tds = CGAL::Triangulation_data_structure_2<CGAL::Periodic_2_triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_vertex_base_2<void> > >, CGAL::Periodic_2_triangulation_face_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_face_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_face_base_2<void> > > >; CGAL::Periodic_2_Delaunay_triangulation_2<Gt, Tds>::Vertex_handle = CGAL::internal::CC_iterator<CGAL::Compact_container<CGAL::Periodic_2_triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_vertex_base_2<CGAL::Triangulation_data_structure_2<CGAL::Periodic_2_triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_vertex_base_2<void> > >, CGAL::Periodic_2_triangulation_face_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_face_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_face_base_2<void> > > > > > >, CGAL::Default, CGAL::Default, CGAL::Default>, false>; CGAL::Periodic_2_Delaunay_triangulation_2<Gt, Tds>::Offset = CGAL::Periodic_2_offset_2]':
/mnt/testsuite/include/CGAL/Periodic_2_Delaunay_triangulation_2.h:1097:40:   required from 'void CGAL::Periodic_2_Delaunay_triangulation_2<Gt, Tds>::remove(CGAL::Periodic_2_Delaunay_triangulation_2<Gt, Tds>::Vertex_handle) [with Gt = CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>; Tds = CGAL::Triangulation_data_structure_2<CGAL::Periodic_2_triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_vertex_base_2<void> > >, CGAL::Periodic_2_triangulation_face_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_face_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_face_base_2<void> > > >; CGAL::Periodic_2_Delaunay_triangulation_2<Gt, Tds>::Vertex_handle = CGAL::internal::CC_iterator<CGAL::Compact_container<CGAL::Periodic_2_triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_vertex_base_2<CGAL::Triangulation_data_structure_2<CGAL::Periodic_2_triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_vertex_base_2<void> > >, CGAL::Periodic_2_triangulation_face_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_face_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_face_base_2<void> > > > > > >, CGAL::Default, CGAL::Default, CGAL::Default>, false>]'
/mnt/testsuite/include/CGAL/Periodic_2_Delaunay_triangulation_2.h:235:23:   required from 'std::ptrdiff_t CGAL::Periodic_2_Delaunay_triangulation_2<Gt, Tds>::insert(InputIterator, InputIterator, bool, typename boost::enable_if<boost::is_convertible<typename std::iterator_traits<_ForwardIterator1>::value_type, typename Gt::Point_2> >::type*) [with InputIterator = __gnu_cxx::__normal_iterator<CGAL::Point_2<CGAL::Epick>*, std::vector<CGAL::Point_2<CGAL::Epick> > >; Gt = CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>; Tds = CGAL::Triangulation_data_structure_2<CGAL::Periodic_2_triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_vertex_base_2<void> > >, CGAL::Periodic_2_triangulation_face_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_face_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_face_base_2<void> > > >; std::ptrdiff_t = long int; typename boost::enable_if<boost::is_convertible<typename std::iterator_traits<_ForwardIterator1>::value_type, typename Gt::Point_2> >::type = void]'
/home/cgal_tester/build/src/cmake/platforms/Debian-Stable/test/Periodic_2_triangulation_2/test_p2t2_delaunay_insert_degenerate.cpp:31:51:   required from here
/mnt/testsuite/include/CGAL/Periodic_2_Delaunay_triangulation_2.h:1139:50: error: template argument for 'template<class T> class boost::thread_specific_ptr' uses local type 'CGAL::Periodic_2_Delaunay_triangulation_2<Gt, Tds>::remove_single_vertex(CGAL::Periodic_2_Delaunay_triangulation_2<Gt, Tds>::Vertex_handle, const Offset&) [with Gt = CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>; Tds = CGAL::Triangulation_data_structure_2<CGAL::Periodic_2_triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_vertex_base_2<void> > >, CGAL::Periodic_2_triangulation_face_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_face_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_face_base_2<void> > > >; CGAL::Periodic_2_Delaunay_triangulation_2<Gt, Tds>::Vertex_handle = CGAL::internal::CC_iterator<CGAL::Compact_container<CGAL::Periodic_2_triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_vertex_base_2<CGAL::Triangulation_data_structure_2<CGAL::Periodic_2_triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_vertex_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_vertex_base_2<void> > >, CGAL::Periodic_2_triangulation_face_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_face_base_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epick>, CGAL::Triangulation_ds_face_base_2<void> > > > > > >, CGAL::Default, CGAL::Default, CGAL::Default>, false>; CGAL::Periodic_2_Delaunay_triangulation_2<Gt, Tds>::Offset = CGAL::Periodic_2_offset_2]::Static_data'
   CGAL_STATIC_THREAD_LOCAL_VARIABLE(Static_data, sd, 30);
                                                  ^
/mnt/testsuite/include/CGAL/tss.h:41:48: note: in definition of macro 'CGAL_STATIC_THREAD_LOCAL_VARIABLE'
        static boost::thread_specific_ptr<TYPE> VAR##_ptr;                   \
                                                ^
/mnt/testsuite/include/CGAL/Periodic_2_Delaunay_triangulation_2.h:1139:50: error:   trying to instantiate 'template<class T> class boost::thread_specific_ptr'
   CGAL_STATIC_THREAD_LOCAL_VARIABLE(Static_data, sd, 30);
                                                  ^
/mnt/testsuite/include/CGAL/tss.h:41:48: note: in definition of macro 'CGAL_STATIC_THREAD_LOCAL_VARIABLE'
        static boost::thread_specific_ptr<TYPE> VAR##_ptr;                   \
                                                ^
```